src/tangent_types/thunks.jl

```diff
@@ -141,7 +145,11 @@ macro thunk(body)
     # Basically `:(Thunk(() -> $(esc(body))))` but use the location where it is defined.
     # so we get useful stack traces if it errors.
     func = Expr(:->, Expr(:tuple), Expr(:block, __source__, body))
```

Comment:
This could go further and generate a name for the thunk, based on source location, instead of anon.  `() ->`. I think the name will be seen sometimes where the code line info is not? 

But not this PR perhaps...